src/init.c

```diff
@@ -841,7 +842,9 @@ static NOINLINE void _finish_julia_init(JL_IMAGE_SEARCH rel, jl_ptls_t ptls, jl_
     JL_TIMING(JULIA_INIT, JULIA_INIT);
     jl_resolve_sysimg_location(rel);
     // loads sysimg if available, and conditionally sets jl_options.cpu_target
-    if (jl_options.image_file)
+    if (rel == JL_IMAGE_IN_MEMORY)
+        jl_set_sysimg_so(jl_RTLD_DEFAULT_handle);
```

Comment:
This is normally an alias for `jl_libjulia_internal_handle`, but I think you wanted to assume `jl_exe_handle` here?

Reply:
Well, I wanted RTLD_DEFAULT, but I'm only testing Linux for now, so it doesn't make a difference. I'll switch this to jl_exe_handle (assuming nothing breaks), but of course we may need extra options in the future potentially.